src/module.c

```diff
@@ -709,13 +740,14 @@ int ReplySeriesRange(RedisModuleCtx *ctx,
         timestamp_t init_ts = (rev == false)
                                   ? series->funcs->GetFirstTimestamp(iterator.currentChunk)
                                   : series->funcs->GetLastTimestamp(iterator.currentChunk);
-        last_agg_timestamp = init_ts - (init_ts % time_delta);
 
+        last_agg_timestamp = init_ts - (init_ts % time_delta) + time_offset;
```

Comment:
Can you explain the logic behind this? i would expect to see the following:
```
        last_agg_timestamp = init_ts - (init_ts % time_delta) + time_offset;
```

Reply:
I'm not sure I follow?

Reply:
sorry pasted the wrong line, i meant to paste this:
```
  last_agg_timestamp = init_ts - ((init_ts + time_offset) % time_delta);
  ```

Reply:
So, I tried that, but it I couldn't get sane aggregations out, and I was struggling to make it work for `RANGE` and `REVRANGE`. 

However; the last time I worked on a C project was about 6 years ago; so I was a little rusty. 

I'm happy to change anything, and double check things on my side!